### PR TITLE
add caching functions

### PR DESCRIPTION
**Pull request by asundaratzenoss**

### CODE REVIEW

**Review by bharatZen — CHANGES_REQUESTED** [review]

**Comment by bharatZen** [comment]
Please add appropriate tests for the compact metrics related changes in the sdk

File: endpoint/endpoint.go [hunk 1]
```diff
@@ -284,6 +292,41 @@ func (e *Endpoint) GetLoggerConfig() log.LoggerConfig {
 	return e.config.LoggerConfig
 }
 
+//GetCacheKey generates key to use for metric whose id is going to be cached
+func (e *Endpoint) GetCacheKey(metric *data_receiver.Metric, tenant string) string {
```
[the remark this answers is not in the document]
Comment (asundaratzenoss) [reply endpoint/endpoint.go:296]: Updated

[the remark this answers is not in the document]
Comment (asundaratzenoss) [reply endpoint/endpoint.go:296]: Updated

File: endpoint/endpoint.go [hunk 2]
```diff
@@ -284,6 +292,41 @@ func (e *Endpoint) GetLoggerConfig() log.LoggerConfig {
...
+	sort.Strings(keys)
+	var flatDims bytes.Buffer
+	for _, k := range keys {
+		flatDims.WriteString(fmt.Sprintf("%s=%s,", k, dims[k]))
+	}
+
+	localKey := fmt.Sprintf("%s:%s:%s", metric.Metric, tenant, flatDims.String())
+	hash := murmur3.New128()
+	hash.Write([]byte(localKey))
+	v1, v2 := hash.Sum128()
```
[the remark this answers is not in the document]
Comment (asundaratzenoss) [reply endpoint/endpoint.go:313]: Updated

File: endpoint/endpoint.go [hunk 3]
```diff
@@ -284,6 +292,41 @@ func (e *Endpoint) GetLoggerConfig() log.LoggerConfig {
...
+	binary.LittleEndian.PutUint32(data, typeNum)
+	binary.LittleEndian.PutUint64(data[4:], v1)
+	binary.LittleEndian.PutUint64(data[12:], v2)
+	return base64.URLEncoding.EncodeToString(data)
+}
+
+//GetMetadataHash generates hash of metadata fields of the metric whose id is going to be cached
+func (e *Endpoint) GetMetadataHash(metric *data_receiver.Metric) string {
+	hasher := murmur3.New64()
+	jm := jsonpb.Marshaler{Indent: "  "}
```
[the remark this answers is not in the document]
Comment (asundaratzenoss) [reply endpoint/endpoint.go]: Updated

**Comment by asundaratzenoss** [comment]
> Please add appropriate tests for the compact metrics related changes in the sdk
Now only calling ConvertMetrics from outside library, so added test for the same

**Review by bharatZen — APPROVED** [review]

**Pull request merged**